workshops/views.py

```diff
@@ -470,7 +470,30 @@ def all_persons(request):
 @login_required
 def person_details(request, person_id):
     '''List details of a particular person.'''
-    person = get_object_or_404(Person, id=person_id)
+    person = Person.objects.filter(
+        id=person_id
+    ).annotate(
+        num_taught=Count(
+             Case(
+                When(task__role__name='instructor', then=Value(1)),
+                output_field=IntegerField()
+            )
+        ),
+        num_helper=Count(
+            Case(
+                When(task__role__name='helper', then=Value(1)),
+                output_field=IntegerField()
+            )
+        ),
+        num_learner=Count(
+            Case(
+                When(task__role__name='learner', then=Value(1)),
+                 output_field=IntegerField()
+            )
+        )
+    ).first()
+    if not person:
+        raise Http404
```

Comment:
Please change this to:

```python
try:
    person = Person.objects.annotate(annotations_here).get(id=person_id)
except Person.DoesNotExist:
    raise Http404('Person matching query does not exist.')
```

I just tested a very simplified annotation and `.get` and it worked, but ensure that these complex annotations work as well.